ansible/configs/sap-hana/files/cloud_providers/ec2_cloud_template.j2

```diff
@@ -0,0 +1,373 @@
+      HostedZoneId: "{{ HostedZoneId }}"
+    {% else %}
+      HostedZoneName: "{{ aws_dns_zone_root }}"
+    {% endif %}
+      RecordSets:
+        - Name: "{{ aws_dns_zone_public }}"
+          Type: NS
+          TTL: {{ aws_dns_ttl_public }}
+          ResourceRecords:
+            "Fn::GetAtt":
+              - DnsZonePublic
+              - NameServers
+    {% endif %}
+
+{% for instance in instances %}
+{% if instance['dns_loadbalancer'] | d(false) | bool
+  and not instance['unique'] | d(false) | bool %}
+  {{instance['name']}}DnsLoadBalancer:
+    Type: "AWS::Route53::RecordSetGroup"
+    DependsOn:
+    {% for c in range(1, (instance['count']|int)+1) %}
+      - {{instance['name']}}{{c}}
+      {% if instance['public_dns'] %}
+      - {{instance['name']}}{{c}}EIP
+      {% endif %}
+    {% endfor %}
+    Properties:
+      {% if secondary_stack is defined %}
+      HostedZoneName: "{{ aws_dns_zone_public }}"
+      {% else %}
+      HostedZoneId:
+        Ref: DnsZonePublic
+      {% endif %}
+      RecordSets:
+      - Name: "{{instance['name']}}.{{aws_dns_zone_public_prefix|d('')}}{{ aws_dns_zone_public }}"
+        Type: A
+        TTL: {{ aws_dns_ttl_public }}
+        ResourceRecords:
+{% for c in range(1,(instance['count'] |int)+1) %}
+          - "Fn::GetAtt":
+            - {{instance['name']}}{{c}}
+            - PublicIp
+{% endfor %}
+{% endif %}
+
+{% for c in range(1,(instance['count'] |int)+1) %}
+  {{instance['name']}}{{loop.index}}:
+    Type: "AWS::EC2::Instance"
+    Properties:
+{% if custom_image is defined %}
+      ImageId: {{ custom_image.image_id }}
+{% else %}
+      ImageId:
+        Fn::FindInMap:
+        - RegionMapping
+        - Ref: AWS::Region
+        - {{ instance.image | default(aws_default_image) }}
+{% endif %}
+      InstanceType: "{{instance['flavor'][cloud_provider]}}"
+      KeyName: "{{instance.key_name | default(key_name)}}"
+    {% if instance['UserData'] is defined %}
+      {{instance['UserData']}}
+    {% endif %}
+
+    {% if instance['security_groups'] is defined %}
+      SecurityGroupIds:
+      {% for sg in instance.security_groups %}
+        - Ref: {{ sg }}
+      {% endfor %}
+    {% else %}
+      SecurityGroupIds:
+        - Ref: DefaultSG
+    {% endif %}
+      SubnetId:
+        Ref: PublicSubnet
+      Tags:
+    {% if instance['unique'] | d(false) | bool %}
+        - Key: Name
+          Value: {{instance['name']}}
+        - Key: internaldns
+          Value: {{instance['name']}}.{{aws_dns_zone_private_chomped}} 
+        - Key: publicname
+          Value: {{instance['name']}}.{{aws_dns_zone_public_prefix|d('')}}{{subdomain_base }}
+    {% else %}
+        - Key: Name
+          Value: {{instance['name']}}{{loop.index}}
+        - Key: internaldns
+          Value: {{instance['name']}}{{loop.index}}.{{aws_dns_zone_private_chomped}}
+        - Key: publicname
+          Value: {{instance['name']}}{{loop.index}}.{{aws_dns_zone_public_prefix|d('')}}{{ subdomain_base}}
```

Comment:
The only difference between your template and the default one is those 2 lines.

Do you really need them ?

If yes, i think we can safely add them to the default template and then you can just remove yours from the config.